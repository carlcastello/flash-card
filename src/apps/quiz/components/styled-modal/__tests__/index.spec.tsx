import React from "react";
import { shallow } from "enzyme";
import { StyledModal } from "../index";
import toJson from "enzyme-to-json";
import { Button, IconButton } from "@material-ui/core";


describe("The Styled Modal", () => {

  let onIgnoreFunction: any;
  let onConfirmFunction: any;

  let wrapper: any;

  beforeEach(() => {
    onIgnoreFunction = jest.fn();
    onConfirmFunction = jest.fn();

    wrapper = shallow(
      <StyledModal
        isOpen={false}
        onConfirm={onConfirmFunction}
        onIgnore={onIgnoreFunction}
        classes={{
          modalContainer: "modal-container",
          modalPaperContainer: "modal-paper-container",
          closeButton: "close-button",
        }}
      />
    )
  });


  describe("the render function", () => {
    it("shows a confirmation modal", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Button)).toHaveLength(2);
    });

    it("show a dialog modal", () => {
      wrapper.setProps({ onConfirm: undefined });
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Button)).toHaveLength(0);
    });
  });

  describe("the onClick", () => {
    it("calls the onIgnoreFunction if IconButton is clicked", () => {
      wrapper.find(IconButton).simulate("click");
      expect(onIgnoreFunction).toHaveBeenCalled();
    }); 

    it("calls the onIgnoreFunction if 'no' Button is clicked", () => {
      wrapper.find(Button).at(0).simulate("click");
      expect(onIgnoreFunction).toHaveBeenCalled();
    }); 

    it("calls the onConfirmFunction if 'yes' Button is clicked", () => {
      wrapper.find(Button).at(1).simulate("click");
      expect(onConfirmFunction).toHaveBeenCalled();
    });
  });
});