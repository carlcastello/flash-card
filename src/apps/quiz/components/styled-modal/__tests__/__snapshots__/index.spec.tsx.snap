// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Styled Modal the render function show a dialog modal 1`] = `
<ForwardRef(Modal)
  className="modal-container"
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(ForwardRef(Paper))
    className="modal-paper-container"
    elevation={3}
  >
    <Styled(MuiBox)
      pb={5}
      position="relative"
      pt={5}
      px={5}
    >
      <WithStyles(ForwardRef(IconButton))
        className="close-button"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</ForwardRef(Modal)>
`;

exports[`The Styled Modal the render function shows a confirmation modal 1`] = `
<ForwardRef(Modal)
  className="modal-container"
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(ForwardRef(Paper))
    className="modal-paper-container"
    elevation={3}
  >
    <Styled(MuiBox)
      pb={2.5}
      position="relative"
      pt={5}
      px={5}
    >
      <WithStyles(ForwardRef(IconButton))
        className="close-button"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-end"
        spacing={1}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            onClick={[MockFunction]}
          >
            No
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            color="secondary"
            onClick={[MockFunction]}
          >
            Yes
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</ForwardRef(Modal)>
`;
